app/assets/javascripts/discourse/controllers/history.js.es6

```diff
@@ -29,6 +29,23 @@ export default Ember.Controller.extend(ModalFunctionality, {
     Discourse.Post.showRevision(postId, postVersion).then(() => this.refresh(postId, postVersion));
   },
 
+  revert(postId, postVersion) {
+    Discourse.Post.revertToRevision(postId, postVersion).then((result) => {
+      this.refresh(postId, postVersion);
+      if (result.topic) {
+        // TO-DO: update topic title here
```

Comment:
Do you not have access to the post object that is being updated somehow? It should have an associated `topic` that you can update if the result contains the new title.

I'd suggest calling `revert(post, postVersion)` then you could say `post.set('topic.title', newTitle)`.